blog/2023-07-18-wing-cloud-launch.md

```diff
@@ -0,0 +1,281 @@
+library only define an abstract API, and it is possible to inject concrete implementations at
+compile time (technically this happens when the Wing preflight code is executed by the compiler).
+This powerful mechanism means that code written against these APIs is **portable across cloud
+providers and provisioning engines**.
+
+And of course, we think of of WCL as the foundation for an ecosystem of cloud abstractions. The
+programming model is open and extensible and anyone will be able to publish libraries and implement
+backends for existing resources based on the needs of their environment.
+
+The WCL is a major project and it's early days. You can see the coverage of resources and targets
+through this [compatibility
+matrix](https://www.winglang.io/docs/standard-library/compatibility-matrix) and make sure to "+1"
+the relevant issue if coverage is missing.
+
+Currently, WCL can only be used from Winglang code, but we are considering releasing it to other
+languages as well. The WCL is built using CDK technologies such as
+[JSII](https://github.com/aws/jsii) and [constructs](https://github.com/aws/constructs), so
+technically it is possible to publish it to other languages.
+
+### Reducing iteration time from minutes to milliseconds
+
+The #1 factor in developer productivity is the time it takes to perform a single iteration
+(sometimes called the *inner dev loop* cycle). This is the time it takes to run my code and check
+that it does what I expected it to do. 
+
+The more an application leans on the cloud, the more this cycle is broken. If for every iteration, I
+need to deploy my code to a personal cloud account, it means that a cycle takes multiple minutes
+(depending on the resources I deploy). Then, having to switch to some other system to run my code
+and trace the logs also increases this iteration time and breaks my development flow.
+
+Some tools are trying to address this problem by "hot swapping" runtime code in the cloud, or by
+proxying requests into my local machine so I can debug locally. But those solutions are only partial as they only take care of the runtime code and not on the entire application. If something changes in my infrastructure, I go back to waiting.
+
+I personally believe that in most cases using real cloud resources for development iterations is
+never going to get us to developer experience nirvana. This is because when we provision resources
+in the cloud, we leverage their [control plane](https://en.wikipedia.org/wiki/Control_plane), which,
+by definition, is not optimized for latency.
+
+In Wing, thanks to the cloud abstraction layer, we are able to take a different approach. The same mechanism we have for cloud portability is used to implement a local cloud simulation target for each resource in the WCL. This means that my entire Wing application can run in a local machine and functionally tested.
+
+### Tests as first-class citizens
```

Comment:
Maybe we could say a few words about the pains of stubbing and mocking (because these are harder / more painful to do in current cloud development because of network latencies and credentials and permissions etc.)